### PR TITLE
rtadvd_test

### PR DESCRIPTION
**Pull request by marjohn56**
Use this patch for testing rtadvd. Please apply this patch after first applying f2dc854, otherwise debug level settings will not work. There are only three debug levels, level 0 is off, 1 is low and 2 is verbose.

### CODE REVIEW

**Comment by fichtner** [comment]
Could be a candidate for 21.1. Happy to leave radvd behind eventually.

**Comment by marjohn56** [comment]
OK, well there's a cock up somewhere I'm trying to sort out, nothing to do with rtadvd, more to do with my patch not applying the way I want it to... trying to apply a patch over a patch for 20.7.4, think I'll bin that idea and and just go with dev, if anyone wants to try it they'll have to switch to dev... what say you?

**Comment by marjohn56** [comment]
Ah.. I'll do it a different way, reverse the first patch and create a whole new one with debug level changes too, that way it will also allow itself to be pulled into 21.1... ,

**Comment by marjohn56** [comment]
@fichtner - Got something weird going on. I have completed ( as far as I can ) all the stuff that creates the rtadvd.conf, added the services so it's separate, all that works and I can start and stop all good. Now the weird bit. If I launch rtadvd from the shell with all the command params, then its working fine, if I launch it from within the GUI, then it shows up on a ps fine, but it will not respond to solicits, from the shell the ps arguments look fine, what's even more strange is that it appears to work fine on my test router, that's just using a single LAN, but on my live router which is using VLANs it doesn't, but as I said, it DOES work fine when launched manually.... very baffled.

**Comment by fichtner** [comment]
Do you use mwexecf()? Try switching to mwexecf_bg() instead.

File: src/etc/inc/plugins.inc.d/dhcpd.inc [hunk 1]
```diff
@@ -535,10 +544,19 @@ function dhcpd_radvd_configure($verbose = false, $blacklist = array())
...
+    if ($config['system']['ipv6_radvd_debug_level'] > 1 ) {
+        $radvddebug = " -D";
+    } else if ($config['system']['ipv6_radvd_debug_level'] = 1 ) {
+        $radvddebug = " -d";
+    } else {
+        $radvddebug = "";
     }
 
+    if (count($radvdifs)) {
+		mwexecf('/usr/sbin/rtadvd %s -c /var/etc/radvd.conf -p /var/run/radvd.pid %s',$radvddebug,$radvd_interfaces);
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/dhcpd.inc:558]: This looks wrong. Second argument is an array with all format arguments.

File: src/etc/inc/plugins.inc.d/dhcpd.inc [hunk 2]
```diff
@@ -535,10 +544,19 @@ function dhcpd_radvd_configure($verbose = false, $blacklist = array())
 
     file_put_contents('/var/etc/radvd.conf', $radvdconf);
 
-    if (count($radvdifs)) {
-        mwexec('/usr/local/sbin/radvd -p /var/run/radvd.pid -C /var/etc/radvd.conf -m syslog');
+    $radvd_interfaces = implode(' ',$rtadvd_interface_list);
```
Comment (fichtner) [inline src/etc/inc/plugins.inc.d/dhcpd.inc:547]: This is wrong too. You implode but later escape a la “int1 int2” which looks like one single interface that doesn’t exist vs. “int1” “int2”

**Comment by fichtner** [comment]
There is also rtadvctl to add/remove interfaces after restart. It looks like a more flexible approach where we don’t need to prime Interfaces at start time.

**Comment by marjohn56** [comment]
Yes, ignore this commit, its just a conversation holder at the moment. All sorts of stuff added now. I'll try the daemon launcher and let you know.. Will be later on tonight as other things to do for a while.

**Comment by marjohn56** [comment]
#4431

**Comment by marjohn56** [comment]
@fichtner - new commits on PR #4431 - This commit is well obsolete!

**Pull request closed without merging**